### PR TITLE
Modernize build system

### LINKED ISSUE #28: Move to single pyproject.toml build
Some things might be easier:

- Currently have requirements in 4 different files (`setup.py`, `requirements.txt`, `docs_requirements.txt` and `test_requirements.txt`) .These could be consolidated as installation options in `pyproject.toml` (for example; lots of ways to do it).
- Would not need the requirements files, `setup.cfg` or `setup.py`, and maybe can also get rid of `conftest.py`

### PR DESCRIPTION
**Pull request by kwinkunks**
Fixes #28. Also updating GitHub workflows, README/CONTRIBUTING, etc.

### CODE REVIEW

**Comment by berland** [comment]
Draft PR or ready for review?

**Comment by kwinkunks** [comment]
@berland Forgot about a dependency, will republish when green

File: .github/workflows/python-sphinx-doc.yml [hunk 1]
```diff
@@ -14,12 +14,11 @@ jobs:
     - name: Set up Python
       uses: actions/setup-python@v5
       with:
-        python-version: '3.x'
+        python-version: '3.11'
```
Comment (berland) [inline .github/workflows/python-sphinx-doc.yml:17]: you don't want to build on the newest, i.e. 3..12?

**Review by berland — APPROVED** [review]
LGTM

**Pull request merged**